File: services/api/app/domain/chat_theme.py
"""Chat themes (ADR-0021 §2.4): five closed slugs, never a free colour.

The slug is all the server stores (`contexts.theme`, CHECKed against this
tuple). The colours live in `packages/shared/tokens.json` under `chatTheme`
for both schemes, and a web-tier test proves every slug here has a full
light and dark palette that clears the contrast floor. The server never sees
a hex value; a colour that is not in tokens cannot be chosen.
"""

from __future__ import annotations

__all__ = ["DEFAULT_THEME", "THEMES", "is_theme"]

THEMES: tuple[str, ...] = ("mac-dinh", "hoang-hon", "bien-dem", "rung-thong", "ruc-ro")
DEFAULT_THEME = "mac-dinh"


def is_theme(value: object) -> bool:
    return isinstance(value, str) and value in THEMES

File: services/core/internal/domain/chattheme/chattheme.go
// Package chattheme is the Go port of app/domain/chat_theme.py (ADR-0021
// §2.4): five closed slugs, never a free colour.
//
// PATCH /contexts/{context_id} refuses any other value with 422
// theme_unknown. The comparison is exact: no case folding, no trimming, no
// normalisation, as Python's `value in THEMES` on a str.
package chattheme

import "slices"

// DefaultTheme is DEFAULT_THEME.
const DefaultTheme = "mac-dinh"

var themes = [...]string{"mac-dinh", "hoang-hon", "bien-dem", "rung-thong", "ruc-ro"}

// Themes is THEMES, in the Python order.
func Themes() []string { return slices.Clone(themes[:]) }

// IsTheme is is_theme for a str.
func IsTheme(value string) bool {
	return slices.Contains(themes[:], value)
}
